### PR TITLE
[12.0][MIG] - contract variable quantity

### PR DESCRIPTION
**Pull request by sbejaoui**

### CODE REVIEW

**Comment by sbejaoui** [comment]
FYI @gva-acsone

**Review by gva-acsone — APPROVED** [review]
Functional test 👍

**Comment by AaronHForgeFlow** [comment]
I am not sure what I am missing. I have a very simple formula in the template: `result=15`

When creating the invoice, the formula changes to result=0. Any ideas?

**Comment by sbejaoui** [comment]
> I am not sure what I am missing. I have a very simple formula in the template: `result=15`
> 
> When creating the invoice, the formula changes to result=0. Any ideas?

Hi @aheficent ,
You are getting this in runbot?

**Comment by AaronHForgeFlow** [comment]
> Hi @aheficent ,
> You are getting this in runbot?

No, in my local. I am debugging into the code and in this line: https://github.com/OCA/contract/blob/e7f5650636233cfeee781e6153171c12c837ccc2/contract_variable_quantity/models/contract_line.py#L38 I get `result=0` as self.qty_formula_id.code

I am not sure if I am missing something

**Comment by sbejaoui** [comment]
> > Hi @aheficent ,
> > You are getting this in runbot?
> 
> No, in my local. I am debugging into the code and in this line:
> 
> https://github.com/OCA/contract/blob/e7f5650636233cfeee781e6153171c12c837ccc2/contract_variable_quantity/models/contract_line.py#L38
> 
> I get `result=0` as self.qty_formula_id.code
> I am not sure if I am missing something

Can you take a look in runbot how I configured the formula and some contracts
I create a formula with `result=line.sequence` and `result=15`.
And it works fine,

**Comment by AaronHForgeFlow** [comment]
@sbejaoui What runbot are you using?

I used this one in this PR: http://3371960-211-e7f565.runbot1-2.odoo-community.org/web

and I got the same thing than in my local. But this must be a different runbot you tested because I did not see any formula result=15 I had to create one

**Comment by sbejaoui** [comment]
> @sbejaoui What runbot are you using?
> 
> I used this one in this PR: http://3371960-211-e7f565.runbot1-2.odoo-community.org/web
> 
> and I got the same thing than in my local. But this must be a different runbot you tested because I did not see any formula result=15 I had to create one

oh I see now, 
you should set `result=15` to code field not to the name.


-->

**Comment by AaronHForgeFlow** [comment]
@sbejaoui haha honestly as the field was not required and the background was white I thought that was part of the explanation and not anc actual editable field haha

**Comment by sbejaoui** [comment]
> @sbejaoui haha honestly as the field was not required and the background was white I thought that was part of the explanation and not anc actual editable field haha

yes, it's confusing :D

**Review by AaronHForgeFlow — APPROVED** [review]
:+1: Functional tests

**Comment by pedrobaeza** [comment]
Can you check Travis status?

File: contract_variable_quantity/models/abstract_contract_line.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Copyright 2016 Tecnativa - Pedro M. Baeza
+# Copyright 2018 Tecnativa - Carlos Dauden
+# Copyright 2018 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class AccountAbstractAnalyticContractLine(models.AbstractModel):
+    _inherit = 'account.abstract.analytic.contract.line'
```
Comment (hparfr) [inline contract_variable_quantity/models/abstract_contract_line.py]: Shoudn't it be     _inherit = 'contract.abstract.contract.line'
 ?

**Review by hparfr — CHANGES_REQUESTED** [review]
From my understanding there should be no dependencies to account.analytic because of the PR #207

File: contract_variable_quantity/models/abstract_contract_line.py (hunk 1 above)
Comment (sbidoul) [reply contract_variable_quantity/models/abstract_contract_line.py]: @hparfr yes, the disconnect from analytics in #207 is fairly recent. It is now complete and needs to be merged then we'll update related PRs.

Comment (pedrobaeza) [reply contract_variable_quantity/models/abstract_contract_line.py]: I'm trying to review it this week.

Comment (hparfr) [reply contract_variable_quantity/models/abstract_contract_line.py]: I did update it quickly https://github.com/sbejaoui/contract/pull/15

**Comment by sbejaoui** [comment]
@OCA/core-maintainers 
This PR can be merged

**Comment by pedrobaeza** [comment]
@sbejaoui please squash together adjacent `OCA Transbot updated translations from Transifex` commits, and also squash your migration commits.

**Comment by sbejaoui** [comment]
> @sbejaoui please squash together adjacent `OCA Transbot updated translations from Transifex` commits, and also squash your migration commits.

Squash done.

**Review by rousseldenis — APPROVED** [review]
Code review. Do you plan to squash some commits?

**Pull request closed without merging**